#! /bin/sh -e
## man-ocamlmklib.dpatch by Samuel Mimram <samuel.mimram@ens-lyon.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: MAnpage for ocamlmklib

if [ $# -ne 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
	exit 1;;
esac

exit 0
@DPATCH@
diff -urN ocaml-3.08.1.orig/man/ocamlmklib.m ocaml-3.08.1/man/ocamlmklib.m
--- ocaml-3.08.1.orig/man/ocamlmklib.m	1970-01-01 01:00:00.000000000 +0100
+++ ocaml-3.08.1/man/ocamlmklib.m	2004-08-19 18:38:55.000000000 +0200
@@ -0,0 +1,127 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.TH OCAMLMKLIB 1 "August 19, 2004"
+.SH NAME
+ocamlmklib \- generate libraries with mixed C / Caml code.
+.SH SYNOPSIS
+.B ocalmklib
+.RI [ options ] " files"
+.SH DESCRIPTION
+The
+.B ocamlmklib
+command facilitates the construction of libraries containing both Caml code and C code, and usable both in static linking and dynamic linking modes.
+.SH OPTIONS
+.TP
+.B \-h, \-\-help
+Show summary of options.
+.TP
+.BI \-cclib\  lib
+C library passed to ocamlc
+.B \-a
+or
+.BR ocamlopt (1)
+.B \-a
+only.
+.TP
+.BI \-ccopt\  opt
+C option passed to
+.BR ocamlc (1)
+.B \-a
+or
+.BR ocamlopt (1)
+.B \-a
+only.
+.TP
+.B \-custom
+Disable dynamic loading.
+.TP
+.BI \-dllpath\  dir
+Add
+.I dir
+to the run-time search path for DLLs.
+.TP
+.BI \-I\  dir
+Add
+.I dir
+to the path searched for Caml object files.
+.TP
+.B \-failsafe
+Fall back to static linking if DLL construction failed.
+.TP
+.BI \-ldopt\ opt
+C option passed to the shared linker only.
+.TP
+.B \-linkall
+Build Caml archive with link-all behavior.
+.TP
+.BI \-l lib
+Specify a dependent C library.
+.TP
+.BI \-L dir
+Add
+.I dir
+to the path searched for C libraries.
+.TP
+.BI \-ocamlc\  cmd
+Use
+.I cmd
+in place of
+.BR ocamlc (1).
+.TP
+.BI \-ocamlopt\  cmd
+Use
+.I cmd
+in place of
+.BR ocamlopt (1).
+.TP
+.BI \-o\  name
+Generated Caml library is named
+.IR name .cma
+or
+.IR name .cmxa.
+.TP
+.BI \-oc\  name
+Generated C library is named
+.RI dll name .so
+or
+.RI lib name .a.
+.TP
+.BI \-rpath\  dir
+Same as
+.B \-dllpath
+.IR dir .
+.TP
+.BI \-R dir
+Same as
+.BR \-rpath .
+.TP
+.B \-verbose
+Print commands before executing them.
+.TP
+\fB\-Wl\fR, \fB\-rpath \fIdir
+Same as
+.B \-dllpath
+.IR dir .
+.TP
+\fB\-Wl\fR, \fB\-rpath\ \-Wl \fIdir
+Same as
+.B \-dllpath
+.IR dir .
+.TP
+\fB\-Wl\fR, \fB\-R\fIdir
+Same as
+.B \-dllpath
+.IR dir .
+.TP
+.BI \-F dir
+Specify a framework directory (MacOSX).
+.TP
+.BI \-framework name
+Use framework
+.I name
+(MacOSX).
+.SH SEE ALSO
+.BR ocamlc (1),
+.BR ocamlopt (1).
+.SH AUTHOR
+This manual page was written by Samuel Mimram <samuel.mimram@ens-lyon.org>,
+for the Debian project (but may be used by others).
